lib/user_space/uthread.c

```diff
@@ -130,15 +130,22 @@ M0_INTERNAL int m0_thread_confine(struct m0_thread *q,
 				  const struct m0_bitmap *processors)
 {
 	size_t    idx;
+	size_t    idx1 = M0_PROCESSORS_INVALID_ID;
 	size_t    nr_bits = min64u(processors->b_nr, CPU_SETSIZE);
 	cpu_set_t cpuset;
 
+	M0_PRE(m0_bitmap_set_nr(processors) > 0);
+
 	CPU_ZERO(&cpuset);
 
 	for (idx = 0; idx < nr_bits; ++idx) {
-		if (m0_bitmap_get(processors, idx))
+		if (m0_bitmap_get(processors, idx)) {
 			CPU_SET(idx, &cpuset);
+			if (idx1 == M0_PROCESSORS_INVALID_ID)
+				idx1 = idx;
+		}
 	}
+	q->t_tls.tls_loci = idx1;
```

Comment:
[* defect *] Wrong level of abstraction. `lib/thread` doesn't know and shouldn't know about the localities. The right solution here would be to move this assignment to `loc_thr_init()`, from where `m0_thread_confine()` is called for locality threads.

Reply:
As @nikitadanilov mentioned elsewhere, historically `m0_thread` knows about "everything". For example, `m0_thread_tls` contains addb data as well as cookie. If we define and document `tls_loci` as "first cpu to which the thread was confined, or 0 if not confined", then it will be the appropriate abstraction level.

I've updated documentation about `tls_loci`, please check.